hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/catalyst/catalog/HoodieCatalogTable.scala

```diff
@@ -212,7 +211,9 @@ class HoodieCatalogTable(val spark: SparkSession, var table: CatalogTable) exten
     } else {
       val (recordName, namespace) = AvroConversionUtils.getAvroRecordNameAndNamespace(table.identifier.table)
       val schema = SchemaConverters.toAvroType(dataSchema, nullable = false, recordName, namespace)
-      val partitionColumns = if (table.partitionColumnNames.isEmpty) {
+      val partitionColumns = if (tableConfigs.contains(KeyGeneratorOptions.PARTITIONPATH_FIELD_NAME.key())) {
+        ConfigUtils.getStringWithAltKeys(tableConfigs.asJava.asInstanceOf[java.util.Map[String, Object]], KeyGeneratorOptions.PARTITIONPATH_FIELD_NAME)
```

Comment:
Similar here on the conversion.

Reply:
I made an attempt to avoid it. I created a new function in ConfigUtils which uses a key mapping function and used that instead in Scala.

Reply:
Synced offline.  We can extract a common config util in Scala and reuse it here.